internal/action/otp.go

```diff
@@ -36,6 +37,48 @@ func (s *Action) OTP(c *cli.Context) error {
 	return s.otp(ctx, name, qrf, clip, pw, true)
 }
 
+func tickingBar(ctx context.Context, cancel context.CancelFunc, expiresAt time.Time, bar *termio.ProgressBar) {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	for tt := range ticker.C {
+		select {
+		case <-ctx.Done():
+			return // returning not to leak the goroutine
+		default:
+			// we don't want to block if not cancelled
+		}
+		if tt.After(expiresAt) {
+			cancel()
+			return
+		}
+		bar.Inc()
+	}
+}
+
+func waitForKeyPress(ctx context.Context, cancel context.CancelFunc) {
+	tty, err := tty.Open()
+	if err != nil {
+		out.Errorf(ctx, "Unexpected error opening tty: %v", err)
+		cancel()
+	}
+	defer tty.Close()
+	for {
+		select {
+		case <-ctx.Done():
+			return // returning not to leak the goroutine
+		default:
+		}
+		r, err := tty.ReadRune()
```

Comment:
Actually I wonder if this is not waiting until the next keystroke, and thus this goroutine isn't returning when it should... (basically it's always in iteration 1 of the for loop...)

Reply:
Yeah, the goroutine is leaking when the OTP Bar finishes on line 52, I think. 
Related: https://benjamincongdon.me/blog/2020/04/23/Cancelable-Reads-in-Go/

Seems like there isn't an easy way out...

Reply:
I think I'll have to refactor to use a cancelable reader, using a `io.Reader` on `os.Stdin` and test it on Windows hoping it will work fine [without all that boilerplate](https://github.com/mattn/go-tty/blob/72ed86c4554d4bf5afd278468f2eb74df8de8f97/tty_windows.go#L200) since we are just caring about key presses...